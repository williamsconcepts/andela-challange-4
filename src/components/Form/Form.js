import React from 'react';

import classes from './Form.module.css';

const form = (props) => {
  const { regionDataChanged, otherInputChanged, clicked } = props;
  return (
    <div className={classes.Form}>
      <h2 className="tc f2">Enter Data</h2>
      <form className="MyForm">
        <div className="div">
          <div>
            <label htmlFor="regionName">Region Name</label>
            <input
              onChange={regionDataChanged}
              required
              data-region-name
              type="text"
              name="name"
              id="regionName"
              placeholder="Example... Africa"
            />
          </div>
          <div>
            <label htmlFor="avgAge">Average Age</label>
            <input
              onChange={regionDataChanged}
              required
              data-avg-age
              type="text"
              name="avgAge"
              id="avgAge"
              placeholder="Example... 19.7"
            />
          </div>
          <div>
            <label htmlFor="avgDailyIncomeInUSD">Average Daily Income In US Dollar</label>
            <input
              onChange={regionDataChanged}
              required
              data-avg-daily-income-in-usd
              type="text"
              name="avgDailyIncomeInUSD"
              id="avgDailyIncomeInUSD"
              placeholder="Example... 5"
            />
          </div>
          <div>
            <label htmlFor="avgDailyIncomePopulation">
              Average Daily Income Population
            </label>
            <input
              onChange={regionDataChanged}
              required
              data-avg-daily-income-population
              type="text"
              name="avgDailyIncomePopulation"
              id="avgDailyIncomePopulation"
              placeholder="Example... 0.71"
            />
          </div>
          <div>
            <label htmlFor="population">Population</label>
            <input
              onChange={otherInputChanged}
              required
              data-population
              type="text"
              name="population"
              id="population"
              placeholder="Example... 66622705"
            />
          </div>
          <div>
            <label htmlFor="timeToElapse">Time to elapse</label>
            <input
              onChange={otherInputChanged}
              required
              data-time-to-elapse
              type="text"
              name="timeToElapse"
              id="timeToElapse"
              placeholder="Example... 58"
            />
          </div>
          <div>
            <label htmlFor="reportedCases">Reported Cases</label>
            <input
              onChange={otherInputChanged}
              required
              data-reported-cases
              type="text"
              name="reportedCases"
              placeholder="Example... 674"
              id="reportedCases"
            />
          </div>
          <div>
            <label htmlFor="totalHospitalBeds">Total Hospital Beds</label>
            <input
              onChange={otherInputChanged}
              required
              data-total-hospital-beds
              type="text"
              name="totalHospitalBeds"
              placeholder="Example... 1380614"
              id="totalHospitalBeds"
            />
          </div>
          <div className="selection">
            <label htmlFor="periodType">Period Type</label>
            <select
              onChange={otherInputChanged}
              required
              data-period-type
              name="periodType"
              id="periodType"
            >
              <option defaultValue value={null}>
                Select Period Type
              </option>
              <option value="days">days</option>
              <option value="weeks">weeks</option>
              <option value="months">months</option>
            </select>
          </div>
        </div>
        <input onClick={clicked} data-go-estimate type="submit" value="Estimate" />
      </form>
    </div>
  );
};

export default form;
