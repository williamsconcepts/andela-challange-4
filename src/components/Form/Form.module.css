.Form {
  display: block;
  width: 80%;
  margin: 2rem auto;
  padding: 3rem;
  background-color: white;
  border-radius: 1rem;
}
.Form form {
  display: flex;
  flex-direction: column;
}

.Form form div {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  column-gap: 2rem;
  row-gap: 2rem;
  font-size: 1.4rem;
  margin: 1rem auto;
}

.Form form div div {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  margin: 0;
}

[data-region-name],
[data-avg-age],
[data-avg-daily-income-in-usd],
[data-avg-daily-income-population],
[data-population],
[data-time-to-elapse],
[data-reported-cases],
[data-total-hospital-beds],
[data-period-type] {
  width: 100%;
  outline: none;
  padding: 0.5rem;
  border-radius: 0.3rem;
  margin: 0.4rem 0 0 0;
  font-style: italic;
}

[data-region-name]:focus,
[data-avg-age]:focus,
[data-avg-daily-income-in-usd]:focus,
[data-avg-daily-income-population]:focus,
[data-population]:focus,
[data-time-to-elapse]:focus,
[data-reported-cases]:focus,
[data-total-hospital-beds]:focus,
[data-period-type]:focus {
  border: 1px solid #0ccac4;
}

[data-go-estimate] {
  align-self: center;
  margin-top: 1rem;
  padding: 1rem 10rem;
  font-size: 1.8rem;
  border-radius: 1rem;
  outline: none;
  color: white;
  border: 1px solid transparent;
  background-color: #54a63b;
  cursor: pointer;
  box-shadow: 1px 1px 10px rgba(0, 0, 0, 0.5);
}

[data-go-estimate]:hover {
  background-color: #4fce28;
}

/* @media (hover: none) and (pointer: coarse) { */
@media (max-width: 1024px) {
  .Form {
    width: 90%;
  }

  .Form form div {
    grid-template-columns: repeat(2, 1fr);
    align-content: center;
  }

  [data-go-estimate] {
    padding: 1rem 8rem;
    font-size: 1.6rem;
  }
}

@media (max-width: 768px) {
  .Form {
    width: 90%;
    padding: 2.5rem;
  }
  .Form form {
    display: flex;
    flex-direction: column;
  }

  .Form form div {
    grid-template-columns: repeat(2, 1fr);
  }
  [data-go-estimate] {
    padding: 1rem 6rem;
    font-size: 1.6rem;
  }
}

@media (max-width: 650px) {
  .Form {
    width: 80%;
    padding: 2rem;
  }

  .Form form div {
    grid-template-columns: 1fr;
    align-content: center;
  }

  [data-go-estimate] {
    /* padding: 1rem 6rem; */
    font-size: 1.6rem;
  }
}

@media (max-width: 500px) {
  .Form {
    width: 100%;
    padding: 2rem;
    border-radius: 0;
  }

  .Form form div {
    grid-template-columns: 1fr;
    align-content: center;
  }

  [data-go-estimate] {
    align-self: stretch;
    padding: 1rem 0rem;
    font-size: 1.6rem;
  }
}
