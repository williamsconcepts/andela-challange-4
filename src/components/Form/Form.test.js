import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import Form from './Form';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Test Form component', () => {
  test('expect to render form component correctly', () => {
    expect.assertions(1);
    expect(shallow(<Form />)).toMatchSnapshot();
  });
});
