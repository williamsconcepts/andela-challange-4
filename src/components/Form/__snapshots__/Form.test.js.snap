// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Form component expect to render form component correctly 1`] = `
<div
  className="Form"
>
  <h2
    className="tc f2"
  >
    Enter Data
  </h2>
  <form
    className="MyForm"
  >
    <div
      className="div"
    >
      <div>
        <label
          htmlFor="regionName"
        >
          Region Name
        </label>
        <input
          data-region-name={true}
          id="regionName"
          name="name"
          placeholder="Example... Africa"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="avgAge"
        >
          Average Age
        </label>
        <input
          data-avg-age={true}
          id="avgAge"
          name="avgAge"
          placeholder="Example... 19.7"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="avgDailyIncomeInUSD"
        >
          Average Daily Income In US Dollar
        </label>
        <input
          data-avg-daily-income-in-usd={true}
          id="avgDailyIncomeInUSD"
          name="avgDailyIncomeInUSD"
          placeholder="Example... 5"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="avgDailyIncomePopulation"
        >
          Average Daily Income Population
        </label>
        <input
          data-avg-daily-income-population={true}
          id="avgDailyIncomePopulation"
          name="avgDailyIncomePopulation"
          placeholder="Example... 0.71"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="population"
        >
          Population
        </label>
        <input
          data-population={true}
          id="population"
          name="population"
          placeholder="Example... 66622705"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="timeToElapse"
        >
          Time to elapse
        </label>
        <input
          data-time-to-elapse={true}
          id="timeToElapse"
          name="timeToElapse"
          placeholder="Example... 58"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="reportedCases"
        >
          Reported Cases
        </label>
        <input
          data-reported-cases={true}
          id="reportedCases"
          name="reportedCases"
          placeholder="Example... 674"
          required={true}
          type="text"
        />
      </div>
      <div>
        <label
          htmlFor="totalHospitalBeds"
        >
          Total Hospital Beds
        </label>
        <input
          data-total-hospital-beds={true}
          id="totalHospitalBeds"
          name="totalHospitalBeds"
          placeholder="Example... 1380614"
          required={true}
          type="text"
        />
      </div>
      <div
        className="selection"
      >
        <label
          htmlFor="periodType"
        >
          Period Type
        </label>
        <select
          data-period-type={true}
          id="periodType"
          name="periodType"
          required={true}
        >
          <option
            defaultValue={true}
            value={null}
          >
            Select Period Type
          </option>
          <option
            value="days"
          >
            days
          </option>
          <option
            value="weeks"
          >
            weeks
          </option>
          <option
            value="months"
          >
            months
          </option>
        </select>
      </div>
    </div>
    <input
      data-go-estimate={true}
      type="submit"
      value="Estimate"
    />
  </form>
</div>
`;
